### PR TITLE
Fix some more OLA RDM test suite discovered issues

### PR DESCRIPTION
**Pull request by peternewman**

### CODE REVIEW

File: src/DMXSerial2.cpp [hunk 1]
```diff
@@ -638,59 +639,76 @@ void DMXSerialClass2::_processRDMMessage(byte CmdClass, uint16_t Parameter, bool
...
+      if (CmdClass == E120_GET_COMMAND) {
+        if (_rdm.packet.DataLength > 0) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          // return the manufacturer label
+          _rdm.packet.DataLength = strlen(_initData->manufacturerLabel);
```
Comment (peternewman) [inline src/DMXSerial2.cpp:684]: This should probably be strnlen, which makes the min redundant.

File: src/DMXSerial2.cpp [hunk 2]
```diff
@@ -638,59 +639,76 @@ void DMXSerialClass2::_processRDMMessage(byte CmdClass, uint16_t Parameter, bool
...
+      if (CmdClass == E120_GET_COMMAND) {
+        if (_rdm.packet.DataLength > 0) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          // return the DEVICE MODEL DESCRIPTION
+          _rdm.packet.DataLength = strlen(_initData->deviceModel);
```
Comment (peternewman) [inline src/DMXSerial2.cpp:704]: Again

File: src/DMXSerial2.cpp [hunk 3]
```diff
@@ -715,23 +733,30 @@ void DMXSerialClass2::_processRDMMessage(byte CmdClass, uint16_t Parameter, bool
           nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
         } else {
           _rdm.packet.DataLength = strlen(deviceLabel);
```
Comment (peternewman) [inline src/DMXSerial2.cpp:735]: Again

**Review by peternewman — COMMENTED** [review]
A potential improvement

File: src/DMXSerial2.cpp (hunk 1 above)
Comment (peternewman) [reply src/DMXSerial2.cpp]: Done

File: src/DMXSerial2.cpp (hunk 3 above)
Comment (peternewman) [reply src/DMXSerial2.cpp]: Done

File: src/DMXSerial2.cpp (hunk 2 above)
Comment (peternewman) [reply src/DMXSerial2.cpp]: Done

**Comment by peternewman** [comment]
If you want to review this @chrisstate, I certainly wouldn't object, although I think these issues have already been fixed in TeensyDMX.

**Comment by peternewman** [comment]
If you want to review this @chrisstaite, I certainly wouldn't object, although I think these issues have already been fixed in TeensyDMX.

**Comment by mathertel** [comment]
Thanks.

**Pull request merged**